### DISCUSSION TITLE
Create copy of an entity?

### DISCUSSION THREAD

**Discussion by stalar** [mikro-orm/mikro-orm#4413] [Q&A] [answered]
Is there a way to create a copy of an entity and then save the copy with a new PK? I want to make a new row in the database with all columns the same except the PK which is auto-incremented.

Something like this:

    const book1 = await em.findOne(Book, 1);
    const book2 = book1.clone();
    await em.flush();
    assert(book2.id);
    assert(book1.id !== book2.id);

Or is there a best practice for achieving the same result?

**Comment by B4nan** [comment, marked as answer]
You could do this:

```ts
const { id, ...data } = wrap(book1).toPOJO();
const book2 = em.create(Book, data);
```
